src/lib/routes/client-api/register.ts

```diff
@@ -61,7 +61,10 @@ export default class RegisterController extends Controller {
         });
     }
 
-    private static resolveEnvironment(user: User, data: Partial<IClientApp>) {
+    private static resolveEnvironment(
+        user: IUser | IApiUser,
```

Comment:
We're already checking `if (user instanceof ApiUser)` below, so this type is more realistic

Reply:
Yes! I feel like we do this loads of places, though 🤔 But you've probably already caught that and fixed it 😄